src/commands/synthetics/run-test.ts

```diff
@@ -77,13 +77,19 @@ export class RunTestCommand extends Command {
         testsToTrigger = await this.getTestsList(api)
       } catch (error) {
         this.reporter.error(`\n${chalk.bgRed.bold(' Failed to get tests list ')}\n${error.message}\n\n`)
-        if (is5xxError(error) && this.config.allowNetworkIssue) {
-          this.reporter.error(`\n${chalk.bgRed.bold(' ERROR on search test endpoint ')}\n`)
+        if (is5xxError(error) && this.config.failOnCriticalErrors) {
```

Comment:
Here and for other conditions: it should be the opposite (`if is5xxError and **NOT** failOnCriticalErrors; then exit with 0; else exit with 1` - with current implementation you'll exit 0 if `failOnCriticalErrors` is `true`)